### PR TITLE
[RISCV][GISel] Support f32/f64 powi.

### PR DESCRIPTION
**Pull request by topperc**
Need to force libcall legalization to treat the integer argument as signed so that it can be promoted to XLen in call lowering for RV64. Alternatively we could promote the operand before converting to libcall, but going through call lowering is closer to what SelectionDAG does.

### CODE REVIEW

**Review by lenary — APPROVED** [review]
LGTM

File: llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp [hunk 1]
```diff
@@ -1290,9 +1290,10 @@ LegalizerHelper::libcall(MachineInstr &MI, LostDebugLocObserver &LocObserver) {
       return UnableToLegalize;
     }
     auto Libcall = getRTLibDesc(MI.getOpcode(), Size);
-    std::initializer_list<CallLowering::ArgInfo> Args = {
+    SmallVector<CallLowering::ArgInfo, 2> Args = {
         {MI.getOperand(1).getReg(), HLTy, 0},
         {MI.getOperand(2).getReg(), ITy, 1}};
+    Args[1].Flags[0].setSExt();
```
Comment (lenary) [inline llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp:1296]: Actually, sorry, I'm not 100% sure about this - should we be using things like `TargetLowering::shouldSignExtendTypeInLibCall`/`TargetLowering::shouldExtendTypeInLibCall` to choose whether to (sign) extend? It's not clear to me we have a GISel equivalent to those callbacks, but I think we might need something like that?

Comment (topperc) [reply llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp:1296]: Eventually yes we probably need something like those hooks. A signed integer argument is maybe the least interesting case for those hooks.

shouldExtendTypeInLibCall defaults to true and only RISC-V and LoongArch override it to return false for FP types.

shouldSignExtendTypeInLibCall defaults to returning true for signed types and only RISC-V, LoongArch, and Mips override it to return true for some unsigned integers.

Comment (lenary) [reply llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp:1296]: Ok, that makes sense

**Review by lenary — APPROVED** [review]
LGTM

**Pull request merged**